GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -905,10 +908,25 @@ private async Task UpdateBranchNameDisplayAsync()
             await TaskScheduler.Default;
 
             var currentBranchName = Module.GetSelectedBranch();
+            var currentBranch = Module.GetRefs(false, true).FirstOrDefault(r => r.LocalName == currentBranchName);
+            string statusBarBranchText;
+            if (string.IsNullOrEmpty(currentBranch.TrackingRemote) || string.IsNullOrEmpty(currentBranch.MergeWith))
+            {
+                statusBarBranchText = string.Format(_statusBarBranchWithoutRemote.Text, currentBranchName);
+            }
+            else
+            {
+                statusBarBranchText = string.Format(_statusBarBranchWithRemote.Text, currentBranchName,
+                    currentBranch.TrackingRemote, currentBranch.MergeWith);
+            }
 
             await this.SwitchToMainThreadAsync();
 
-            branchNameLabel.Text = currentBranchName;
+            branchNameLabel.Click += (object sender, EventArgs e) =>
```

Comment:
Subscribtion could have been made more than once (when you create a branch) that have for consequences that the remote form is open multiple times (also fixed in my branch)